src/main/java/com/google/devtools/build/lib/rules/platform/PlatformRule.java

```diff
@@ -52,14 +54,24 @@ public RuleClass build(RuleClass.Builder builder, RuleDefinitionEnvironment env)
         .add(
             attr(CONSTRAINT_VALUES_ATTR, BuildType.LABEL_LIST)
                 .allowedFileTypes(FileTypeSet.NO_FILE)
-                .mandatoryProviders(ImmutableList.of(ConstraintValueInfo.PROVIDER.id())))
+                .mandatoryProviders(ConstraintValueInfo.PROVIDER.id()))
 
         /* <!-- #BLAZE_RULE(platform).ATTRIBUTE(remote_execution_properties) -->
         A string used to configure a remote execution platform. Actual builds make no attempt to
         interpret this, it is treated as opaque data that can be used by a specific SpawnRunner.
         <!-- #END_BLAZE_RULE.ATTRIBUTE --> */
         .add(attr(REMOTE_EXECUTION_PROPS_ATTR, Type.STRING))
 
+        /* <!-- #BLAZE_RULE(platform).ATTRIBUTE(parents) -->
+        The label of a <code>platform</code> target that this platform should inherit from. Although
+        the attribute takes a list, there should be no more than one platform present. Any
+        constraint_settings not set directly on this platform will be found in the parent platform.
```

Comment:
Did we have API reference to the precise semantics? (what happens if both platforms define the constraint_setting, {PARENT_REMOTE_EXECUTION_PROPERTIES}, etc)?

Reply:
This is the API reference, I can add more detail if anything is unclear.

Reply:
For the sake of precision, can we clarify that the child value overrides the parent value if both are set (even if that's intuitive anyway)? Is it worth mentioning the use of {PARENT_REMOTE_EXECUTION_PROPERTIES} here?

This seems like the right place o be pedantically detailed.

Reply:
Looks great!